Author: Benjamin Lowell <lowell.bv@gmail.com>
Date:   Wed Jan 22 02:08:25 2025 -0400

	This patch addresses the issue of duplex printing not working as intended on Solus by setting Duplex Long edge to use `&l1S`.
	The reason this patch is needed is that only the PLC duplex settings are honored, and `Duplex rotated` in `brlaser.drv` is ignored.
	
	PCL Command Overview:
	&l2S: Duplex Short edge
	&l1S: Duplex Long  edge
	&l0S: Simplex (one sided)

	This change has not been applied upstream for the following reasons:
	- When duplex support was added to `brlaser`, printing Duplex Long edge on the `DCP-7065DN` resulted in missing lines.
	- The workaround at the time involved always printing Short edge when duplexing, and rotating the reverse side by 180 degrees for Long edge printing.
	- It is unknown if the missing lines issue is still present when using `&l1S` for Duplex Long edge on select printers.

	Technically, `Duplex rotated` should be replaced with `Duplex normal` (in `brlaser.drv.in`), even though this change will not have a functional impact.  
	Setting `Duplex normal` should avoid any further disruption if CUPS starts respecting the duplex settings in `brlaser.drv` at some point in the future.

diff --git a/src/job.cc b/src/job.cc
index 9cc5e35..bb3b25b 100644
--- a/src/job.cc
+++ b/src/job.cc
@@ -78,8 +78,10 @@ void job::write_page_header() {
   fputs("\033E", out_);
   fprintf(out_, "\033&l%dX", std::max(1, page_params_.num_copies));
 
-  if (page_params_.duplex || page_params_.tumble) {
+  if (page_params_.tumble) {
     fputs("\033&l2S", out_);
+  } else if (page_params_.duplex) {
+    fputs("\033&l1S", out_);
   } else {
     fputs("\033&l0S", out_);
   }
